parachain/pallets/ethereum-beacon-client/src/tests.rs

```diff
@@ -788,7 +915,7 @@ mod beacon_minimal_tests {
 			ValidatorsRoot::<mock_minimal::Test>::set(get_validators_root::<mock_minimal::Test>());
 			LatestFinalizedHeaderState::<mock_minimal::Test>::set(FinalizedHeaderState {
 				beacon_block_root: H256::default(),
-				beacon_slot: update.block.slot,
+				beacon_slot: update.block.slot - 1,
```

Comment:
Hmmm I still don't understand how this would work because the finalized header slot will always be greater than the header update slot?

Reply:
Yes, seems a big mistake here, let me fix it first then we continue the review